src/diagnostic-channel-publishers/src/azure-coretracing.pub.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+import { channel, IModulePatcher, PatchFunction } from "diagnostic-channel";
+import * as coreTracingTypes from "@azure/core-tracing";
+
+export const DEFAULT_TRACER_TAG = "__AzureMonitorTracer__";
+
+/**
+ * By default, @azure/core-tracing default tracer is a NoopTracer.
+ * This patching changes the default tracer to a patched BasicTracer
+ * which emits ended spans as diag-channel events.
+ * @param coreTracing
+ */
+const azureCoreTracingPatchFunction: PatchFunction = (coreTracing: typeof coreTracingTypes) => {
+    try {
+        const BasicTracer = require('@opentelemetry/tracing').BasicTracer;
```

Comment:
doesn't BasicTracer support the SpanProcessor interface? Why do you need to override tracer.startSpan?

Reply:
Good point. My current thinking is that I am unsure if I want to just use the OT tracer that pre-made is already made or define my own Tracer class here that implements `coretracing.Tracer` and maintain it myself. If I use the OT tracer, I don't have to maintain it and can use span processors, but then I'll need to require users to install that package to get these patches working (I don't want to include it as a default dependency). This still seems better than the alternative.

Reply:
I have no real preference. Not sure how much work maintaining your own tracer would be. Depends on the expected drift from OT. 😉